src/components/UpdateVote.jsx

```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import { useState } from "react";
+
+function UpdateVote({ vote, setVote, setArticle, article }) {
+  const [liked, setLiked] = useState(false);
+  const [unLiked, setUnLiked] = useState(false);
+  const [likeDisable, likeSetDisable] = useState(false);
+  const [unLikeDisable, unLikeSetDisable] = useState(false);
+  function upVote(e) {
+    if (liked) {
+      axios
+        .patch(
+          `https://nc-news-pa.herokuapp.com/api/articles/${article.article_id}`,
+          { inc_votes: -1 }
+        )
+        .then((res) => {
+          setVote(res.data.votes);
+        });
+
+      setVote((current) => {
+        let newVote = current - 1;
+
```

Comment:
a lot of repeated code here - try to keep your code DRY

If the differences in these functions is `+1` or `-1` then consider passing it in as a variable and having one function that handles both